### DISCUSSION TITLE
change middleware on every request

### DISCUSSION THREAD

**Discussion by reharik** [prisma/prisma#9750] [Q&A]
Hi, I have/need middleware that accepts the current logged in user and updates the "createById" and "updatedById" properties of every write.  I have just realized that the middleware, which lives in the single instance of prismaclient is a stack and each time I "add" the middleware with the current user it's adding to that stack.  
It seem like a current user specific hook would be a fairly common need.  For instance I have/need another middleware that makes sure that a user can only act upon or view data from it's own organization.  Thus each request needs to pass the current users orgId to the middleware. 
Any help with this would be appreciated.
Thanks,
R

**Comment by ryands17** [comment]
@reharik 👋 

There's a similar request here so maybe you add the above comment there and also a 👍 for the request: https://github.com/prisma/prisma/issues/6882

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
